runtime/vm/createramclass.cpp

```diff
@@ -2301,13 +2301,16 @@ internalCreateRAMClassDone(J9VMThread *vmThread, J9ClassLoader *classLoader, J9C
 		}
 
 #if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
-		if (J9ROMCLASS_ALLOWS_NON_ATOMIC_CREATION(romClass)) {
+	if (J9_ARE_ALL_BITS_SET(romClass->optionalFlags, J9_ROMCLASS_OPTINFO_PRELOAD_ATTRIBUTE)) {
```

Comment:
The flag to be checked should be `J9_ROMCLASS_OPTINFO_IMPLICITCREATION_ATTRIBUTE`.

Reply:
woops. thanks!